HIP/hip-xxx.md

```diff
@@ -0,0 +1,921 @@
+                    "0000000000000000000000000000000000000000000000000000000000000080"
+                ],
+                "storage": null,
+                "reason": null
+            },
+            {
+                "pc": 548,
+                "op": "POP",
+                "gas": 2308,
+                "gasCost": 2,
+                "depth": 1,
+                "stack": [
+                    "00000000000000000000000000000000000000000000000000000000094f6253",
+                    "00000000000000000000000000000000000000000000000000000000000000a5",
+                    "0000000000000000000000008fa74f0a79112c8c46c8e6739eb8cc929083298c",
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                    "0000000000000000000000000000000000000000000000000000000000000014",
+                    "0000000000000000000000000000000000000000000000000000000000000014"
+                ],
+                "memory": [
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                    "0000000000000000000000000000000000000000000000000000000000000080"
+                ],
+                "storage": null,
+                "reason": null
+            },
+            [...],
+        ]
+    }
+}           
+```
+
+### Mirror node Trace REST API
+
+The JSON-RPC Relay will in turn call the Mirror Node REST API. Depending on the `tracer` that was provided as an input parameter to the `debug_traceTransaction` RPC endpoint, one of two Mirror node endpoints will be called:
+1. For `debug_traceTransaction` called with `callTracer`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/actions` endpoint will be called.
+2. For `debug_traceTransaction` called with `opcodeLogger`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/opcodes` endpoint will be called.
+
+The `/actions` endpoint is already implemented. 
+
+This section will specify the `/opcodes` endpoint.
+
+#### `/api/v1/contracts/results/{transactionIdOrHash}/opcodes`
+
+**Input parameters:** The endpoint will accept three optional parameters:
+
+```
+// Boolean flag controlling whether to return stack information. Defaults to `true`.
+
+stack: <bool> 
+
+// Boolean flag controlling whether to return memory information. Defaults to `false`.
+
+memory: <bool> 
+
+// Boolean flag controlling whether to return storage information. Defaults to `false`.
+
+storage: <bool>
+```
+
+Disabling stack, memory and storage traces will significantly reduce the size of the opcode related data. This would be helpful for reducing the network bandwidth. 
+
+**Output**: detailed information for the transaction execution with `stack`, `memory` and `storage` set to `true`
+Example output:
+
+```
+{
+    "gas": 52139,
+    "failed": false,
+    "returnValue": "0x0000000000000000000000000000000000000000000000000000000000000001",
```

Comment:
There's a lot of concerns with the API that I would push to address if we can never change it again.

* The REST API always uses snake_case for field names.
* The primary array field `structLogs` should be named after the endpoint `opcodes`
* Should include `contract_id`
* Considering pagination again

Are we intending this just as a disabled by default, only intended for passthrough consumption via the relay sort of API? And are we okay with breaking it later if that scope increases and we want to enable it by default?

Reply:
I believe the idea is to disable the /opcodes endpoint by default since it could be very memory and computation heavy API. It could be enabled by the user on a local instance, but stays disabled on production environments, controlling this via a feature flag. We could design it in such a way, that if a paid provider services wants to maintain Mirror Archive Node with /opcodes API - they could just change the feature flag and enable it for users.

In any case, I like the idea of pagination, since output could be hundreds of MBs.